lib/process-services-cloud/src/lib/services/web-socket.service.spec.ts

```diff
@@ -0,0 +1,140 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { Apollo, gql } from 'apollo-angular';
+import { of, Subject } from 'rxjs';
+import { WebSocketService } from './web-socket.service';
+import { SubscriptionOptions } from '@apollo/client/core';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { AuthenticationService, AppConfigService } from '@alfresco/adf-core';
+import { FeaturesServiceToken } from '@alfresco/adf-core/feature-flags';
+
+describe('WebSocketService', () => {
+    let service: WebSocketService;
+    const onLogoutSubject: Subject<void> = new Subject<void>();
+
+    const apolloMock = jasmine.createSpyObj('Apollo', ['use', 'createNamed']);
+    const isOnSpy = jasmine.createSpy('isOn$');
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [HttpClientTestingModule],
+            providers: [
+                {
+                    provide: Apollo,
+                    useValue: apolloMock
+                },
+                {
+                    provide: AppConfigService,
+                    useValue: {
+                        get: () => 'wss://testHost'
+                    }
+                },
+                {
+                    provide: AuthenticationService,
+                    useValue: {
+                        getToken: () => 'testToken',
+                        onLogout: onLogoutSubject.asObservable()
+                    }
+                },
+                {
+                    provide: FeaturesServiceToken,
+                    useValue: {
+                        isOn$: isOnSpy
+                    }
+                }
+            ]
+        });
+        service = TestBed.inject(WebSocketService);
+        TestBed.inject(FeaturesServiceToken);
+        apolloMock.use.and.returnValues(undefined, { subscribe: () => of({}) });
+        isOnSpy.and.returnValues(of(true));
+    });
+
+    afterEach(() => {
+        apolloMock.use.calls.reset();
+        apolloMock.createNamed.calls.reset();
+    });
+
+    it('should not create a new Apollo client if it is already in use', (done) => {
+        const apolloClientName = 'testClient';
+        const subscriptionOptions: SubscriptionOptions = { query: gql(`subscription {testQuery}`) };
+        const wsOptions = { apolloClientName, wsUrl: 'testUrl', subscriptionOptions };
+
+        apolloMock.use.and.returnValues(true, { subscribe: () => of({}) });
+
+        service.getSubscription(wsOptions).subscribe(() => {
```

Comment:
instead of subscribing and use the done function we can transform this into a Promise and then await for it, it makes the test more readable